lib/web_ui/test/engine/semantics/accessibility_test.dart

```diff
@@ -43,76 +55,89 @@ void testMain() {
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'polite message');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
+    void expectNoMessages() => expectMessages();
+
+    test('Default value of aria-live is polite when assertiveness is not specified', () async {
+      accessibilityAnnouncements.handleMessage(codec, encodeMessageOnly(message: 'polite message'));
+      expectMessages(polite: 'polite message');
+
+      await Future<void>.delayed(liveMessageDuration);
+      expectNoMessages();
     });
 
-    test('aria-live is assertive when assertiveness is set to 1', () {
-      resetAccessibilityAnnouncements();
-      const Map<dynamic, dynamic> testInput = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'assertive message', 'assertiveness': 1}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, '');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, 'assertive message');
+    test('aria-live is assertive when assertiveness is set to 1', () async {
+      sendAnnouncementMessage(message: 'assertive message', assertiveness: 1);
+      expectMessages(assertive: 'assertive message');
+
+      await Future<void>.delayed(liveMessageDuration);
+      expectNoMessages();
     });
 
-    test('aria-live is polite when assertiveness is null', () {
-      resetAccessibilityAnnouncements();
-      const Map<dynamic, dynamic> testInput = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'polite message', 'assertiveness': null}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'polite message');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
+    test('aria-live is polite when assertiveness is null', () async {
+      sendAnnouncementMessage(message: 'polite message');
+      expectMessages(polite: 'polite message');
+
+      await Future<void>.delayed(liveMessageDuration);
+      expectNoMessages();
     });
 
-    test('aria-live is polite when assertiveness is set to 0', () {
-      resetAccessibilityAnnouncements();
-      const Map<dynamic, dynamic> testInput = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'polite message', 'assertiveness': 0}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'polite message');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
+    test('aria-live is polite when assertiveness is set to 0', () async {
```

Comment:
Do we know/care what happens if assertiveness is a negative int? Or is this not a scenario we'll run into